Artifacts/windows-vsts-build-agent/vsts-agent-install.ps1

```diff
@@ -0,0 +1,122 @@
+# Downloads the Visual Studio Online Build Agent, installs on the new machine, registers with the Visual
+# Studio Online account, and adds to the specified build agent pool
+
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory=$true)]
+    $vstsAccount,
+    
+    [Parameter(Mandatory=$true)]
+    $vstsUserPassword,
+    
+    [Parameter(Mandatory=$true)]
+    $agentName,
+    
+    [Parameter(Mandatory=$true)]
+    $poolname,
+    
+    [Parameter(Mandatory=$true)]
+    $driveLetter
+)
+
+if ($driveLetter.length -ne 1)
+{
+    Write-Error "The drive letter must be 1 character only; exiting."
+    exit -1
+}
+
+if ($vstsAccount -match "https*://" -or $vstsAccount -match "visualstudio.com")
+{
+    Write-Error "VSTS account should not be the URL, just the account name."
+    exit -1
+}
+
+$vstsUser = "AzureDevTestLabs"
+
+$currentLocation = Split-Path -parent $MyInvocation.MyCommand.Definition
+
+# Create a temporary directory to download from VSTS the agent package (agent.zip) to, and then launch the configuration.
+$agentTempFolderName = Join-Path $env:temp ([System.IO.Path]::GetRandomFileName())
+New-Item -ItemType Directory -Force -Path $agentTempFolderName
+
+$serverUrl = "https://$vstsAccount.visualstudio.com"
+
+$vstsAgentUrl = "$serverUrl/_apis/distributedtask/packages/agent"
+
+$retryCount = 3
+$retries = 1
+do
+{
+    try
+    {
+        $basicAuth = ("{0}:{1}" -f $vstsUser, $vstsUserPassword) 
```

Comment:
Actually that doesn't work. The first character in a formatted string like that can't be a variable.